File: TypeScript/mediaOO.ts
class Aluno {
    Nome: string;
    NotaG1: number;
    NotaG2: number;
    Media: number;

    ///Construtor
    constructor(name: string) {
        this.Nome = name;
    }
    
    ///seta nota do G1
    SetG1(nota: number) {
        this.NotaG1 = nota;
    }

    ///seta nota do G2
    SetG2(nota: number) {
        this.NotaG2 = nota;
    }

    ///calcula a média
    CalcMedia() {
        this.Media = (this.NotaG1 + this.NotaG2 * 2 ) / 3;
    }

    ///Mostra resultado do aluno
    ShowResult() {
        var result = this.Media >= 6 ? "aprovado" : "reprovado";
        var textHtml = "Aluno " + this.Nome + ", " + result + " com a média final " + this.Media;
        let p = document.createElement('p');
        p.textContent = textHtml;

        document.body.appendChild(p);
    }
}

//Aqui começa o programa
let aluno = new Aluno("Roberto");

aluno.SetG1(3.5);
aluno.SetG2(8);
aluno.CalcMedia();

aluno.ShowResult();

File: python/mediaOO.py
class Aluno:
    def __init__(self, name):
        self.Nome = name
        
    def SetG1(self, nota):
        self.G1 = nota
        
    def SetG2(self, nota):
        self.G2 = nota
    
    def CalcMedia(self):
        self.Media = (self.G1 + self.G2 * 2)/3
        
    def Print(self):
        print("Aluno: ", self.Nome)
        print("Media final: %.2f" % (self.Media))
        
        if self.Media > 6:
            print("Resultado: Aluno APROVADO")
        else:
            print("Resultado: Aluno REPROVADO")

##Aqui começa o programa de fato
continuar = 1
while continuar != 0:
    aluno = Aluno(input ("informe o nome do aluno: "))
    aluno.SetG1(float(input ("Informe do G1: ")))
    aluno.SetG2(float(input ("Informe do G2: ")))

    aluno.CalcMedia()
    aluno.Print()
    
    print("1 - Verificar outro aluno")
    print("0 - Sair")
    continuar = int(input ("Selecione uma opção: "))